cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureMetadataCollector.java

```diff
@@ -129,9 +130,76 @@ private String getLocalityIndicator(Integer faultDomainCount, CloudContext cloud
+            LOGGER.debug("Attempting to collect metadata for load balancer {}, type {}", loadBalancerName, type);
+            try {
+                String ip = null;
+                if (LoadBalancerType.PUBLIC.equals(type)) {
+                    ip = lookupPublicIp(resourceGroup, azureClient, loadBalancerName);
+                } else if (LoadBalancerType.PRIVATE.equals(type)) {
+                    ip = lookupPrivateIp(resourceGroup, azureClient, loadBalancerName);
+                }
+
+                if (StringUtils.isNotEmpty(ip)) {
+                    CloudLoadBalancerMetadata loadBalancerMetadata = new CloudLoadBalancerMetadata.Builder()
+                        .withType(type)
+                        .withIp(ip)
+                        .withName(loadBalancerName)
+                        .build();
+                    cloudLoadBalancerMetadata.add(loadBalancerMetadata);
+                    LOGGER.debug("Saved metadata for load balancer: {}", loadBalancerMetadata);
+                } else {
+                    LOGGER.warn("Unable to find metadata for load balancer {}.", loadBalancerName);
+                }
+            } catch (RuntimeException e) {
+                LOGGER.warn("Unable to find metadata for load balancer " + loadBalancerName, e);
+            }
+        }
+
+        return cloudLoadBalancerMetadata;
     }
 
+    private String lookupPrivateIp(String resourceGroup, AzureClient azureClient, String loadBalancerName) {
+        String ip = null;
+        List<String> privateIps = azureClient.getLoadBalancerIps(resourceGroup, loadBalancerName, LoadBalancerType.PRIVATE);
+        if (privateIps.isEmpty()) {
+            LOGGER.warn("Unable to find private ip address for load balancer {}", loadBalancerName);
+        } else {
+            ip = privateIps.get(0);
+            if (privateIps.size() > 1) {
+                LOGGER.warn("Found multiple private IPs ({}) for load balancer {}. Only one, {}, will be used.",
+                        String.join(", ", privateIps),
+                        loadBalancerName,
+                        ip);
+            }
+        }
+        return ip;
+    }
+
+    private String lookupPublicIp(String resourceGroup, AzureClient azureClient, String loadBalancerName) {
+        String ip = null;
+        List<String> publicIps = azureClient.getLoadBalancerIps(resourceGroup, loadBalancerName, LoadBalancerType.PUBLIC);
+        if (publicIps.isEmpty()) {
+            LOGGER.warn("Unable to find public ip address for load balancer {}", loadBalancerName);
+        } else {
+            ip = publicIps.get(0);
+            if (publicIps.size() > 1) {
+                LOGGER.warn("Found multiple public IPs ({}) for load balancer {}. Only one, {}, will be used.",
+                        String.join(", ", publicIps),
+                        loadBalancerName,
+                        ip);
+            }
+        }
+        return ip;
+    }
```

Comment:
I've added refactoring this to [CB-12229](https://jira.cloudera.com/browse/CB-12229).

I think Haley and I discussed the issue with multiple IPs being returned, and settled on just using one since that one is persisted and then treated as the canonical IP moving forward.

There's the possibility of the IPs potentially changing after we provision, if that's a realistic concern I think we should open a ticket for an improvement and think carefully about when it would happen.

Reply:
I think we've a misunderstanding of the issue here. 
My concern is, if for any reasons azure returns multiple IPs, how can we ensure we always choose the same? Is it ordered on azure side? shall we order somehow and choose the first? Is it even a valid problem?

Reply:
I don't think we have a good way to pick any particular IP address out of a set if more than one is returned.

Given that, I think I'll just sort the `List<String>` lexicographically. They're ip addresses, so the sort isn't particular meaningful. It should at least ensure a stable selection with `get(0)` unless IP addresses are added or removed.